packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -9,98 +9,319 @@ import 'package:file/file.dart';
+    final File goldenFile = getGoldenFile(golden);
     if (!goldenFile.existsSync()) {
       throw TestFailure('Could not be compared against non-existent file: "$golden"');
     }
-    final List<int> goldenBytes = await goldenFile.readAsBytes();
-    // TODO(tvolkert): Improve the intelligence of this comparison.
-    if (goldenBytes.length != imageBytes.length) {
-      return false;
-    }
-    for (int i = 0; i < goldenBytes.length; i++) {
-      if (goldenBytes[i] != imageBytes[i]) {
+
+    if (platform.isLinux) {
+      final List<int> goldenBytes = await goldenFile.readAsBytes();
+      if (goldenBytes.length != imageBytes.length) {
         return false;
       }
+      for (int i = 0; i < goldenBytes.length; i++) {
+        if (goldenBytes[i] != imageBytes[i]) {
+          return false;
+        }
+      }
+      return true;
     }
+    print('Skipping "$golden" : Skia Gold unavailable && !isLinux');
     return true;
   }
 
+  /// Updates the golden file identified by [golden] with [imageBytes].
+  ///
+  /// This will be invoked in lieu of [compare] when [autoUpdateGoldenFiles]
+  /// is `true` (which gets set automatically by the test framework when the
+  /// user runs `flutter test --update-goldens`).
+  @override
+  Future<void> update(Uri golden, Uint8List imageBytes) async {
+    final File goldenFile = getGoldenFile(golden);
+    await goldenFile.parent.create(recursive: true);
+    await goldenFile.writeAsBytes(imageBytes, flush: true);
+  }
+
+  /// Returns the golden file identified by the given [Uri].
+  @override
+  File getGoldenFile(Uri uri) {
+    return fs.directory(basedir).childFile(fs.file(uri).path);
+  }
+
+  /// Modifies the given [Uri] to incorporate an optional [version] number.
+  ///
+  /// Golden files for package:flutter use version numbers, and must be used to
+  /// reference the appropriate file.
+  @override
+  Uri getTestUri(Uri key, int version) {
+    return _addVersion(key, version);
+  }
+
+  Uri _addVersion(Uri key, int version) {
+    final String extension = path.extension(key.toString());
+    return version == null ? key : Uri.parse(
+      key
+        .toString()
+        .split(extension)
+        .join() + '.' + version.toString() + extension
+    );
+  }
+}
+
+/// Implementation of the abstract base class [FlutterGoldenFileComparator] for
+/// testing against with Skia Gold.
+///
+/// For testing across all platforms, the [SkiaGoldClient] is used to upload
+/// images for framework-related golden tests and process results. Currently
+/// these tests are designed to be run post-submit on Cirrus CI, informed by the
+/// environment.
+class FlutterSkiaGoldFileComparator implements FlutterGoldenFileComparator {
+  /// Creates a [FlutterSkiaGoldFileComparator] that will resolve golden file
+  /// URIs relative to the specified [basedir].
+  ///
+  /// The [fs] parameter exists for testing purposes only.
+  @visibleForTesting
+  FlutterSkiaGoldFileComparator(
+    this.basedir, {
```

Comment:
Looks like these three properties exist in both sub-classes. Should they be moved to the superclass?